src/lib/ngsiNotify/QueueNotifier.cpp

```diff
@@ -186,5 +187,23 @@ void QueueNotifier::sendNotifyContextRequest
     return;
   }
 
+  extern int notifAlarmThreshold;
+
+  if (notifAlarmThreshold != 0)
+  {
+    std::string details = ("notification queue reached maximum threshold");
+
+    long unsigned int threshold = queueSize(service)*notifAlarmThreshold/100;
+
+    if (threshold >= queueSize(service))
+    {
+      alarmMgr.notificationQueue(queueName.c_str(), details.c_str());
+    }
+  }
```

Comment:
Code can be simplified this way:

```suggestion
    long unsigned int threshold = queueSize(service)*notifAlarmThreshold/100;

    if (threshold >= queueSize(service))
    {
      alarmMgr.notificationQueue(queueName.c_str(), "notification queue reached maximum threshold");
    }
  }
```